runtime/expr/eval.go

```diff
@@ -706,7 +706,7 @@ func (i *Index) Eval(ectx Context, this *zed.Value) *zed.Value {
 	}
 }
 
-func indexArray(zctx *zed.Context, ectx Context, typ *zed.TypeArray, array zcode.Bytes, index *zed.Value) *zed.Value {
+func indexVector(zctx *zed.Context, ectx Context, inner zed.Type, b zcode.Bytes, index *zed.Value) *zed.Value {
```

Comment:
Nit: Stick with the convention of giving the `zcode.Bytes` parameter a meaningful name.
```suggestion
func indexVector(zctx *zed.Context, ectx Context, inner zed.Type, vector zcode.Bytes, index *zed.Value) *zed.Value {
```